nautobot/extras/context_managers.py

```diff
@@ -232,7 +253,14 @@ def web_request_context(
 
             # TODO: get_snapshots() currently requires a DB query per object change processed.
             # We need to develop a more efficient approach: https://github.com/nautobot/nautobot/issues/6303
-            snapshots = oc.get_snapshots()
+            snapshots = oc.get_snapshots(
+                pre_object_data[str(oc.changed_object_id)]
+                if pre_object_data is not None and str(oc.changed_object_id) in pre_object_data
+                else None,
```

Comment:
Could we default `pre_object_data` (and `_v2` to `{}` instead of `None`, and then simplify this to:

```suggestion
                pre_object_data.get(str(oc.changed_object_id), None),
```